tests/unit/test_signature.cpp

```diff
@@ -0,0 +1,427 @@
+      0x0D};
+
+const std::vector<uint8_t> SignatureTest::_validPKCS1SignatureSHA512{
+      0x77, 0x10, 0x46, 0xCC, 0xDF, 0xCC, 0x45, 0xD7, 0xEF, 0xE0, 0xC6, 0x19, 0xF4, 0xD1, 0x4C,
+      0xC9, 0x55, 0x31, 0xC5, 0x4B, 0xEB, 0x82, 0xC3, 0xC8, 0x89, 0x1D, 0x33, 0x24, 0xE8, 0xDA,
+      0x0F, 0x40, 0x01, 0xC7, 0x8C, 0x1D, 0x25, 0x84, 0x6E, 0x46, 0x7C, 0xBB, 0x0E, 0xAA, 0x24,
+      0xAF, 0x86, 0xBC, 0x3A, 0xB7, 0x98, 0x32, 0x71, 0x17, 0xC4, 0x45, 0xD7, 0x06, 0xF5, 0x30,
+      0xAF, 0x7D, 0x50, 0xA4, 0xA1, 0xC5, 0x67, 0x71, 0x17, 0x7D, 0xF1, 0x0D, 0xC1, 0xEA, 0x02,
+      0x69, 0x6C, 0x46, 0x70, 0x4A, 0x88, 0x16, 0x8C, 0xF5, 0xEB, 0xAB, 0x74, 0x3F, 0x6E, 0x99,
+      0x15, 0x37, 0x60, 0x2D, 0x6A, 0xA1, 0x03, 0x55, 0x31, 0x3B, 0x3A, 0xA4, 0xB2, 0x38, 0x16,
+      0xC1, 0xDC, 0xDF, 0xB9, 0xAF, 0x56, 0x1D, 0xCB, 0xE9, 0xAC, 0x46, 0x9A, 0xA7, 0x48, 0x04,
+      0x5C, 0x2D, 0x15, 0x04, 0x76, 0x26, 0xDE, 0xB2, 0x89, 0xDC, 0x06, 0xF5, 0xAF, 0xA6, 0x6F,
+      0xDD, 0x72, 0x05, 0x8F, 0x12, 0xFF, 0x6A, 0x6A, 0x5C, 0x2E, 0xAD, 0x09, 0xAD, 0xCD, 0x20,
+      0x68, 0x8B, 0x7C, 0xC8, 0x98, 0x30, 0xB1, 0x0F, 0xA1, 0x73, 0x75, 0x92, 0xDB, 0x32, 0x32,
+      0x82, 0x7A, 0x5A, 0x65, 0x46, 0x49, 0xE1, 0x8D, 0x23, 0xB9, 0x2E, 0x99, 0xA4, 0xEA, 0x57,
+      0x5F, 0x1F, 0x6A, 0xC7, 0x56, 0x3E, 0x5C, 0x04, 0x44, 0xCC, 0xB0, 0x66, 0xB1, 0x6E, 0xB1,
+      0x33, 0x09, 0x4F, 0x52, 0x03, 0x78, 0x56, 0x12, 0x28, 0x2D, 0xD5, 0x38, 0x3F, 0x48, 0x8F,
+      0xC8, 0x64, 0x8B, 0xEA, 0xD6, 0xFF, 0xD1, 0xEF, 0xD3, 0xE7, 0x10, 0x48, 0xF3, 0x44, 0x3E,
+      0x48, 0x6A, 0xCE, 0xC4, 0xD0, 0x0A, 0x26, 0x38, 0x17, 0xFE, 0x3D, 0x7B, 0xBE, 0x82, 0x83,
+      0xFE, 0x98, 0x59, 0x01, 0x35, 0xBB, 0xB6, 0x60, 0xF7, 0xAA, 0x0E, 0x90, 0xF7, 0xDE, 0xA6,
+      0xAA};
+
+const std::vector<uint8_t> SignatureTest::_testMessageDigestSHA256{
+      0x7F, 0x83, 0xB1, 0x65, 0x7F, 0xF1, 0xFC, 0x53, 0xB9, 0x2D, 0xC1,
+      0x81, 0x48, 0xA1, 0xD6, 0x5D, 0xFC, 0x2D, 0x4B, 0x1F, 0xA3, 0xD6,
+      0x77, 0x28, 0x4A, 0xDD, 0xD2, 0x00, 0x12, 0x6D, 0x90, 0x69};
+
+const std::vector<uint8_t> SignatureTest::_testMessageDigestSHA512{
+      0x86, 0x18, 0x44, 0xD6, 0x70, 0x4E, 0x85, 0x73, 0xFE, 0xC3, 0x4D, 0x96, 0x7E,
+      0x20, 0xBC, 0xFE, 0xF3, 0xD4, 0x24, 0xCF, 0x48, 0xBE, 0x04, 0xE6, 0xDC, 0x08,
+      0xF2, 0xBD, 0x58, 0xC7, 0x29, 0x74, 0x33, 0x71, 0x01, 0x5E, 0xAD, 0x89, 0x1C,
+      0xC3, 0xCF, 0x1C, 0x9D, 0x34, 0xB4, 0x92, 0x64, 0xB5, 0x10, 0x75, 0x1B, 0x1F,
+      0xF9, 0xE5, 0x37, 0x93, 0x7B, 0xC4, 0x6B, 0x5D, 0x6F, 0xF4, 0xEC, 0xC8};
+
+/**
+ * @brief Tests whether the interfaces for creating a signature are all outputing the correct
+ *        PKCS1 signature by comparing it to a result known to be correct.
+ * Cases:
+ * - Sign a plain text message using a signature context
+ * - Sign a message digest using a signature context
+ * - Sign a plain text message using a private key
+ * - Sign a message digest using a private key
+ */
+TEST_F(SignatureTest, testSignaturePKCS1)
```

Comment:
It is better to have smaller, focused and well isolated tests (see comment below)